lib/soapy_cake/client.rb

```diff
@@ -18,16 +18,34 @@ def run(request)
       request.api_key = api_key
       request.time_offset = time_offset
 
-      response = Response.new(http_response(request), request.short_response?)
-      response.time_offset = time_offset
-      response.to_enum
+      if request.opts[:response].present?
+        local_response(request)
+      else
+        http_response(request)
+      end.public_send(:"to_#{xml_response? ? 'xml' : 'enum'}")
     end
 
     private
 
+    def xml_response?
+      opts[:xml_response] == true
+    end
+
     def http_response(request)
       url = "https://#{domain}#{request.path}"
-      HTTParty.post(url, headers: headers, body: request.xml, timeout: NET_TIMEOUT).body
+      http_response = HTTParty.post(url, headers: headers, body: request.xml, timeout: NET_TIMEOUT)
+
+      fail RequestFailed, "Request failed with HTTP #{http_response.code}: " \
+        "#{http_response.body}" unless http_response.success?
+      response = Response.new(http_response, request.short_response?)
```

Comment:
This is duplicated, why not just `local_response(http_response)`? (maybe with renaming the method a bit...)

Also, having a variable with the same name as the method is confusing.